File: php/features.php
<?php
declare(strict_types=1);

// EmailValidation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EmailValidationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EmailValidationBaseFeature();
            case "test":
                return new EmailValidationTestFeature();
            default:
                return new EmailValidationBaseFeature();
        }
    }
}

File: py/features.py
# EmailValidation SDK feature factory

from feature.base_feature import EmailValidationBaseFeature
from feature.test_feature import EmailValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EmailValidationBaseFeature(),
        "test": lambda: EmailValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
